File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        ListNode strt=head ,end=head;
        while(end!=null && end.next !=null){
            strt=strt.next;
            end=end.next.next;
        }
        ListNode prev=null , curr=strt;
        while(curr!=null){
            ListNode temp=curr.next;
            curr.next=prev;
            prev=curr;
            curr=temp;
        }
        ListNode i=head,j=prev;
        while(j!=null){
            if(i.val!=j.val){
                return false;
            }
            i=i.next;
            j=j.next;
        }
        return true;
    }
}

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        sta=[]
        curr=head
        while curr:
            sta.append(curr.val)
            curr=curr.next
        curr=head
        while curr:
            if sta.pop()!=curr.val:
                return False
            curr=curr.next
        return True